tests/proxy/replicated_map/replicated_map_test.go

```diff
@@ -0,0 +1,345 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package replicated_map
+
+import (
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/core"
+	"github.com/hazelcast/hazelcast-go-client/internal/common"
+	. "github.com/hazelcast/hazelcast-go-client/rc"
+	. "github.com/hazelcast/hazelcast-go-client/tests"
+	"log"
+	"reflect"
+	"sort"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+)
+
+var rmp core.ReplicatedMap
+var client hazelcast.IHazelcastInstance
+
+func TestMain(m *testing.M) {
+	remoteController, err := NewRemoteControllerClient("localhost:9701")
+	if remoteController == nil || err != nil {
+		log.Fatal("create remote controller failed:", err)
+	}
+	cluster, err := remoteController.CreateCluster("3.9", DEFAULT_XML_CONFIG)
+	remoteController.StartMember(cluster.ID)
+	client, _ = hazelcast.NewHazelcastClient()
+	rmp, _ = client.GetReplicatedMap("myReplicatedMap")
+	m.Run()
+	rmp.Clear()
+	client.Shutdown()
+	remoteController.ShutdownCluster(cluster.ID)
+}
+
+func TestReplicatedMapProxy_Name(t *testing.T) {
+	name := "myReplicatedMap"
```

Comment:
no need for this variable. In fact, eliminating the variable would make the test more readable.

Reply:
@mustafaiman  I think these tests are not necessary actually.(Name,ServiceName,PartitionKey). They are already tested here https://github.com/hazelcast/hazelcast-go-client/blob/master/tests/proxy/map/map_test.go#L53. Do you think they are necessary ?

Reply:
Agree, not really neccessary.